snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -774,9 +788,13 @@ def clean_stage(self, project_staged_state):
 
         self.mark_cleaned(steps.STAGE)
 
+    def _do_runner_step(self, step: steps.Step):
```

Comment:
You can also put self.makedirs in here too (pull, stage and prime call that and should be ok for build)

Reply:
will do